File: quivi.guests.react.app/src/hooks/api/Dtos/orders/Order.ts
import type { OrderExtraCost } from "./OrderExtraCost";
import type { OrderItem } from "./OrderItem";
import type { OrderState } from "./OrderState";
import type { OrderType } from "./OrderType";

export interface OrderChangeLog {
    readonly state: OrderState;
    readonly note: string;
}

export interface OrderFieldValue {
    readonly id: string;
    readonly value: string;
}

export interface Order {
    readonly id: string;
    readonly sequenceNumber: string;
    readonly scheduledTo?: string;
    readonly merchantId: string;
    readonly channelId: string;
    readonly state: OrderState;
    readonly type: OrderType;
    readonly items: OrderItem[];
    readonly lastModified: string;
    readonly extraCosts: OrderExtraCost[];
    readonly changes: OrderChangeLog[];
    readonly fields: OrderFieldValue[];
}

File: Quivi.Guests.Api/Dtos/Order.cs
namespace Quivi.Guests.Api.Dtos
{
    public enum OrderState
    {
        Draft = -2,
        Requested = 0,
        Scheduled = 1,
        Rejected = 2,
        Processing = 3,
        Completed = 4,
    }

    public enum OrderType
    {
        OnSite = 0,
        TakeAway = 1,
    }

    public enum ExtraCostType
    {
        None = 0,
        SurchargeFee = 1,
        Tip = 2,
    }

    public class Order
    {
        public required string Id { get; init; }
        public required string MerchantId { get; init; }
        public required string SequenceNumber { get; init; }
        public required string ChannelId { get; init; }
        public OrderState State { get; init; }
        public OrderType Type { get; init; }
        public required IEnumerable<OrderItem> Items { get; init; }
        public DateTimeOffset? ScheduledTo { get; init; }
        public DateTimeOffset LastModified { get; init; }
        public required IEnumerable<OrderExtraCost> ExtraCosts { get; init; }
        public required IEnumerable<OrderChangeLog> Changes { get; init; }
        public required IEnumerable<OrderFieldValue> Fields { get; init; }
    }
}